src/digraph.rs

```diff
@@ -939,6 +962,113 @@ impl PyDiGraph {
         Ok(index.index())
     }
 
+    /// Find node within this graph given a specific weight
+    ///
+    /// :param obj: The weight to look for
+    ///
+    /// returns: the index > 0 if a node was found and otherwise -1
+    /// rtype: int
+    pub fn find_node_by_weight(
+        &self,
+        py: Python,
+        obj: PyObject,
+    ) -> Option<usize> {
+        let mut index = None;
+        for node in self.graph.node_indices() {
+            let weight = self.graph.node_weight(node).unwrap();
+            let weight_compare = |a: &PyAny, b: &PyAny| -> PyResult<bool> {
+                let res = a.compare(b)?;
+                Ok(res == Ordering::Equal)
+            };
+
+            if weight_compare(obj.as_ref(py), weight.as_ref(py)).unwrap() {
+                index = Some(node.index());
+                break;
+            }
+        }
+        index
+    }
+
+    ///     Returns the graph that results from contracting u and v.
+    /// Node contraction identifies the two nodes as a single node incident to any edge that was incident to the original two nodes.
+    /// Parameters:
+    /// G (NetworkX graph) – The graph whose nodes will be contracted.
+    /// v (u,) – Must be nodes in G.
+    /// self_loops (Boolean) – If this is True, any edges joining u and v in G become self-loops on the new node in the returned graph.
+    /// Returns:
+    /// A new graph object of the same type as G (leaving G unmodified) with u and v identified in a single node. The right node v will be merged into the node u, so only u will appear in the returned graph.
+    /// Return type:
+    /// Networkx graph
+
+    /// Merge two nodes in the graph.
+    ///
+    /// :param node0: The source node that is going to be merged
+    /// :param node1: The target node that is going to be the new node
+    #[text_signature = "(u, v /)"]
+    pub fn merge_nodes(
+        &mut self,
+        py: Python,
+        u: usize,
+        v: usize,
+    ) -> PyResult<()> {
+        let source_node = NodeIndex::new(u);
+        let target_node = NodeIndex::new(v);
```

Comment:
Just a small docs nit here. I probably would include a description of the behavior of merging. I also would make node0 and node1 consistent vs `u` and `v`. Like:

```suggestion
    /// Merge two nodes in the graph.
    ///
    /// If the nodes have equal weight objects then all the edges into and out of `node0`` will be added
    /// to ``node1`` and ``node0`` will be removed from the graph. If the nodes don't have equal weight
    /// objects then no changes will be made and no error raised
    ///
    /// :param int node0: The source node that is going to be merged
    /// :param int node1: The target node that is going to be the new node
    #[text_signature = "(node0, node1, /)"]
    pub fn merge_nodes(
        &mut self,
        py: Python,
        node0: usize,
        node1: usize,
    ) -> PyResult<()> {
        let source_node = NodeIndex::new(node0);
        let target_node = NodeIndex::new(node1);
```

But feel free to adjust that as you see fit. That's just the gist of what I think is missing.

Reply:
fixed and thanks for this